tests/test_dataset.py

```diff
@@ -530,3 +550,157 @@ def test_get_or_add_mag_for_tiff():
     layer.get_or_add_mag("1")
     assert "1" in layer.mags.keys()
     assert mag.name == "1"
+
+
+def test_tiled_tiff_read_and_write_multichannel():
+    delete_dir("./testoutput/TiledTiffDataset")
+    tiled_tiff_ds = TiledTiffDataset.create(
+        "./testoutput/TiledTiffDataset",
+        scale=(1, 1, 1),
+        tile_size=(32, 32),
+        pattern="{xxx}_{yyy}_{zzz}.tif",
+    )
+
+    mag = tiled_tiff_ds.add_layer("color", "color", num_channels=3).add_mag("1")
+
+    data = np.zeros((3, 250, 250, 10), dtype=np.uint8)
+    for h in range(10):
+        for i in range(250):
+            for j in range(250):
+                data[0, i, j, h] = i
+                data[1, i, j, h] = j
+                data[2, i, j, h] = 100
+
+    mag.write(data, offset=(5, 5, 5))
+    written_data = mag.read(size=(250, 250, 10), offset=(5, 5, 5))
+    assert written_data.shape == (3, 250, 250, 10)
+    assert np.array_equal(data, written_data)
+
+
+def test_tiled_tiff_read_and_write():
+    delete_dir("./testoutput/tiled_tiff_dataset")
+    tiled_tiff_ds = TiledTiffDataset.create(
+        "./testoutput/tiled_tiff_dataset",
+        scale=(1, 1, 1),
+        tile_size=(32, 32),
+        pattern="{xxx}_{yyy}_{zzz}.tif",
+    )
+
+    mag = tiled_tiff_ds.add_layer("color", "color").add_mag("1")
+
+    data = np.zeros((250, 250, 10), dtype=np.uint8)
+    for h in range(10):
+        for i in range(250):
+            for j in range(250):
+                data[i, j, h] = i + j % 250
+
+    mag.write(data, offset=(5, 5, 5))
+    written_data = mag.read(size=(250, 250, 10), offset=(5, 5, 5))
+    assert written_data.shape == (1, 250, 250, 10)
+    assert np.array_equal(written_data, np.expand_dims(data, 0))
+
+    assert mag.get_tile(1, 1, 6).shape == (1, 32, 32, 1)
+    assert np.array_equal(
+        mag.get_tile(1, 1, 6)[0, :, :, 0],
+        TiffReader("./testoutput/tiled_tiff_dataset/color/1/001_001_006.tif").read(),
+    )
```

Comment:
I'd add another assertion which also compares the the tiff content against `data[32 * 1: 32 * 2, 32 * 1: 32 * 2, 6]`. Right now, you are only testing that `get_tile` reads the correct tiff, but we should also test that the correct data was written to that tiff.

Additionally, I'd change the tested tile index of (1, 1, 6) so something asymmetric (e.g., (1, 2, 6)).